src/Components/Components/src/ILazyLoader.cs

```diff
@@ -0,0 +1,15 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Threading.Tasks;
+
+namespace Microsoft.AspNetCore.Components
+{
+    public interface ILazyLoader
```

Comment:
Since this is about loading assemblies, could this be called `IAssemblyLoader` or even something like `IWebAssemblyLazyLoader`?

From a layering point of view, it would be great to have this in the `M.A.C.WebAssembly` package since it's only intended to be used there. Maybe there's some reason why that's not possible - will be looking out while reviewing the rest of the code here :)

Reply:
I agree, I would be fine with IWebAssemblyAssemblyLoader, IAssemblyLoader is too familiar with AssemblyLoader which is a .NET Core concept, and I'm not a fan of having "Lazy" show up on the public API surface.

Reply:
This interface is used in the both the WASMAssemblyLoader on the client and the ServerAssemblyLoader for pre-rendering scenarios. I hesitated to call it `IWebAssemblyAssemblyLoader` for this reason.

Sidenote: This is also why this common interface is under the M.A.C namespace.

I think `ILazyAssemblyLoader` is fine. We already use `Lazy` in the name of the MSBuild property used to designate lazy-loaded assemblies so we've bit the bullet on using it as part of the public API.